pkg/deploy/spindeploy/transformer/defaults_test.go

```diff
@@ -0,0 +1,90 @@
+package transformer
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestConfig_SetArchaiusDefaults(t *testing.T) {
+	s := `
+apiVersion: spinnaker.io/v1alpha2
+kind: SpinnakerService
+metadata:
+  name: spinnaker
+spec:
+  spinnakerConfig:
+    profiles:
+      gate: {}
+`
+	tr, spinsvc := th.setupTransformerFromSpinText(&defaultsTransformerGenerator{}, s, t)
+	before_ := spinsvc.GetSpinnakerConfig().Profiles["gate"]
+	before := *before_.DeepCopy()
+	err := tr.TransformConfig(context.TODO())
+	assert.Nil(t, err)
+
+	config := spinsvc.GetSpinnakerConfig()
+	gate := config.Profiles["gate"]
+	assert.NotNil(t, gate)
```

Comment:
I think this needs to check that all services have a profile section with `archaius: true`, not just gate.